src/ServiceControl/Recoverability/Retrying/RetryProcessor.cs

```diff
@@ -195,6 +207,59 @@ async Task<int> Stage(RetryBatch stagingBatch, IAsyncDocumentSession session)
             return messages.Length;
         }
 
+        async Task TryStageMessage(FailedMessage message, string stagingId, FailedMessageRetry failedMessageRetry)
+        {
+            try
+            {
+                await StageMessage(message, stagingId).ConfigureAwait(false);
+            }
+            catch (Exception e)
+            {
+                if (failedMessageRetry.StageAttempts < MaxStagingAttempts)
+                {
+                    Log.Error($"Attempt {failedMessageRetry.StageAttempts + 1} of {MaxStagingAttempts} to stage a retry message {message.UniqueMessageId} failed", e);
```

Comment:
Pedantic now, but perhaps a Warn here and an error when breaking out of the retry attempts?

Reply:
Fixed